cdap-common/src/main/java/co/cask/cdap/common/kerberos/ImpersonationRequest.java

```diff
@@ -27,10 +28,27 @@
 public class ImpersonationRequest {
   private final NamespacedEntityId entityId;
   private final ImpersonatedOpType impersonatedOpType;
+  private final String principal;
+  private final String keytabURI;
 
+  // used when we do not know what is the principal for the entity id
   public ImpersonationRequest(NamespacedEntityId entityId, ImpersonatedOpType impersonatedOpType) {
+    this(entityId, impersonatedOpType, null);
+  }
+
+  // principal is needed for remote side to make query to master but we should not pass the key tab uri
+  public ImpersonationRequest(NamespacedEntityId entityId, ImpersonatedOpType impersonatedOpType,
+                              @Nullable String principal) {
```

Comment:
It can be null from other constructors I think, there is a constructor which only takes entityId and impersonationOpType.